README.md

```diff
@@ -1 +1,99 @@
-# js-datastore-pubsub
\ No newline at end of file
+# datastore-pubsub
+
+[![](https://img.shields.io/badge/made%20by-Protocol%20Labs-blue.svg?style=flat-square)](http://protocol.ai)
+[![](https://img.shields.io/badge/project-IPFS-blue.svg?style=flat-square)](http://ipfs.io/)
+[![](https://img.shields.io/badge/freenode-%23ipfs-blue.svg?style=flat-square)](http://webchat.freenode.net/?channels=%23ipfs)
+[![standard-readme](https://img.shields.io/badge/standard--readme-OK-green.svg?style=flat-square)](https://github.com/RichardLitt/standard-readme)
+[![js-standard-style](https://img.shields.io/badge/code%20style-standard-brightgreen.svg?style=flat-square)](https://github.com/feross/standard)
+
+> provides an implementation of an interface-datastore compliant api for pubsub
```

Comment:
As this aims to be interface-datastore compliant, we should also run those tests against this, like https://github.com/ipfs/js-datastore-fs/blob/master/test/index.spec.js#L130-L155.

Reply:
I completely agree with you and tried to add those tests. 

But after that, I realized some important aspects. There are several mandatory functions which are not needed, such as `check`, `open`, `batch`, etc . Moreover, `put` function will have a different end result compared with `datastore-fs`, for example, as it will publish the pubsub topic, in order to store the record received in all the subscribers datastore, instead of storing in its own datastore.

After some thinking, I think that the end game of this module should be to be compliant with [TieredDatastore](https://github.com/ipfs/js-datastore-core/blob/master/src/tiered.js#L18) instead of adding too much unneeded complexity. 

For making this possible, I would say that `interface-datastore` could have a set of tests related specifically with `TieredDatastore` and this module should run those tests. However, `TieredDatastore` also have some functions which are not needed for this module, at least for its current requirements (`IPNS over Pubsub`).

All things considered, I suggest that if you agree, we should change the current documented end game of this module being compliant with `interface-datastore`. Then, create issues on `interface-datastore` + `js-datastore-pubsub` to create the means to allow us to guarantee that this module will be compliant with `TieredDatastore`. This way, we will not block `IPNS over Pubsub` for now and we, or someone in the community, may get to this effort later, in order to not delay the OKRs. This way, we can even extend more the discussion in those issues, if this should be entirely compliant with `interface-datastore`, or just with `TieredDatastore`. 

Let me know what do you think @jacobheun

Reply:
If this is to be used with `TieredDatastore` as one of its stores, if the methods aren't implemented this would risk calls to those methods throwing not a function errors. It might not make sense to fully implement the method for the pubsub datastore, but it shouldn't throw errors. At the very least it would seem we'd need stubbed methods with logs and immediate callbacks. If this is to be used completely separate from `TieredDatastore` then it's probably not needed, but that doesn't seem to be the case.

@vasco-santos is there a sample implementation of this yet?

Reply:
Yes, I agree to add all the methods throwing the not implement error.

It is intended to be used with `TieredDatastore` as one of its stores.

I have a WIP commit using it [here](https://github.com/ipfs/js-ipfs/commit/4c36183c80cb20b563e4f73901f855304f429934#diff-8d1a7002931f907b006489a19a204fd8R92)